sqs.tf

```diff
@@ -34,7 +36,20 @@ POLICY
 }
 
 resource "aws_sns_topic_subscription" "sqs_hive_metastore_sns_subscription" {
-  topic_arn = "${var.metastore_events_sns_topic}"
-  protocol  = "sqs"
-  endpoint  = "${aws_sqs_queue.shuntingyard_sqs_queue.arn}"
+  topic_arn     = "${var.metastore_events_sns_topic}"
+  protocol      = "sqs"
+  endpoint      = "${aws_sqs_queue.shuntingyard_sqs_queue.arn}"
+  filter_policy = "${data.template_file.sqs_hive_metastore_sns_subscription_filter.rendered}"
+}
+
+data "template_file" "sqs_hive_metastore_sns_subscription_filter" {
+  template = <<EOF
+  {
+    "qualifiedTableName": [ $${tables_list} ]
+  }
+EOF
+
+  vars {
+    tables_list = "${join(",", formatlist("\"%s\"", var.selected_tables))}"
+  }
 }
```

Comment:
This block should be moved above `aws_sns_topic_subscription` block for better readability.